### PR TITLE
Process Uploaded Video File Locally

### LINKED ISSUE #418: GSoC Infragram.org full-screen UI and video upload Discussion and Planning
Hi, everyone! This is the Planning Issue for the [Infragram.org: full-screen UI and video upload](https://summerofcode.withgoogle.com/programs/2022/projects/rP8hthBg) for [infragram.org](https://infragram.org/). The project is in collaboration with the [Outreachy proposal: Design a new full-screen user interface for Infragram.org](https://publiclab.org/notes/stefq1111/04-08-2022/outreachy-proposal) . All are open for discussion and improvement.

**Process Video Locally**
- [x] Allow conversion of entire video.
- [x] Implement video controls on canvas.
- [x] Include another Canvas Element ( second Canvas ).
- [x] Record  video on process at second canvas.
- [x]  Save processed video file.
- [x] Integrate the above functionalities to with UI of outreachy collaborator. @stephaniequintana 

**Allow selecting multiple output resolutions as well**
- [x] Dynamically change the size of second canvas to fit required resolution.
- [x] Integrate the above functionalities to with UI of outreachy collaborator.

**Integrate Teamsys latest camera selection API** 
 Issue #39
- [ ] Download Temasys/AdapterJS library.
- [ ] Change all instances of the word `webRtcOptions.context ` to `webRtcDetectedType`
- [ ] Remove code `if (webRtcOptions.context === "flash")`
[Reference 1](https://github.com/publiclab/spectral-workbench.js/pull/172)  [Reference 2]( https://github.com/publiclab/spectral-workbench.js/issues/87)

**Accept multiple webcam resolutions**
- [x] Maintain css canvas size.
- [x] Change HTML canvas size to fit requested resolution.
- [x] Integrate the above functionalities to with UI of outreachy collaborator.

**Write UI tests for new Infragram UI Jest** 
- [ ] write test for infragram UI functionalities.

**Allow zooming/panning of video**
- [x]  Zooming :  Add uScale of uniform vect2 to shader.frag and shader.vert. 
- [ ]  Panning :  Add uPan of uniform vect2 to shader.frag and shader.vert .
- [ ] Implement above functionalities to infragram drawscreen function.
- [ ] Integrate the above functionalities to with UI of outreachy collaborator.

**Miscellaneous**

- [ ]  Add option to close webcam input
- [ ]  Enable image and video stream mirroring
- [x]  Add file filters on input to optimize infrared analysis

### PR DESCRIPTION
**Pull request by Forchapeatl**
**Process Uploaded Video File Locally** 
Fixes #418  
**Passed Test case**
- Upload Video.
- Switch between video and Webcam
- Switch between video  and image file.
- Switch between multiple videos.
- Switch between multiple image files

Found at [https://forchapeatl.github.io/infragram/indexs.html#](https://forchapeatl.github.io/infragram/indexs.html#)

**Switch between image and video file**

https://user-images.githubusercontent.com/24577149/184059853-9b4e034e-78c5-434b-80c5-e42675504728.mp4

**Switch between camera and video file**

https://user-images.githubusercontent.com/24577149/184056709-09aed545-e499-4d8e-a766-d5b510e3e060.mp4

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [X] PR is descriptively titled 📑 and links the original issue above 🔗
* [X] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
* [X] code is in uniquely-named feature branch and has no merge conflicts 📁
* [X] screenshots/GIFs are attached 📎 in case of UI updation
* [X] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

**Comment by Forchapeatl** [comment]
Hello @stephaniequintana , Here is the clean PR for video upload and processing. During the drag and drop please drop the video file as `id= "localVideo"`  and append  it under this element . Thank you. 

https://github.com/publiclab/infragram/blob/550a2d26def4fe2d29c0baf77d1c50034565fdfb/index2.html#L42

**Comment by Forchapeatl** [comment]
@jywarren , @TildaDares  please for your feedback on this PR

File: index2.html [hunk 1]
```diff
@@ -22,7 +22,7 @@
   <script src="node_modules/webrtc-adapter/out/adapter.js"></script>
   <script src="node_modules/getusermedia-js/dist/getUserMedia.min.js"></script>
 	
-  <script src="dist/infragram2.js"></script>
+  <script src="dist/infragram.js"></script>
```
Comment (jywarren) [inline index2.html:25]: Very cool! So do we need to keep dist/infragram2.js now? Just curious, thanks for all the thorough testing! I'd love to hear from @stephaniequintana on this as well, just a review would be helpful! Thank you!

Comment (Forchapeatl) [reply index2.html:25]: I think there's no need to keep the dist/infragram2.js file.In the mean time please permit me move forward with the merge.

Comment (Forchapeatl) [reply index2.html:25]: @stephaniequintana, your review is very important on this topic. Please I will consider and apply it on the next PR. For now please  let me merge this.

**Pull request merged**